test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -0,0 +1,178 @@
+package pao_mustgather
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	mustGatherPath = "../../testdata/must-gather/must-gather.e2e"
+)
+
+var mustGatherContentDir string
+var snapshotDir string
+var snapshotName string
+
+var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", func() {
+	var profile *performancev2.PerformanceProfile
+
+	testutils.BeforeAll(func() {
+		mustgatherPathContent, err := ioutil.ReadDir(mustGatherPath)
+		Expect(err).To(BeNil(), "failed to read the Mustgather Directory %s: %v", mustGatherPath, err)
+
+		for _, directory := range mustgatherPathContent {
+			if strings.Contains(directory.Name(), "must-gather") {
+				mustGatherContentDir = filepath.Join(mustGatherPath, directory.Name())
+			}
+		}
+
+		// pre generated must-gather data
+		// hardcored node name
+		// TODO genearate data in live cluster
```

Comment:
this TODO is still valid ?  do we need to keep it ?

Reply:
I'm not a fan of adding TODOs in the code. Crossreference PRs and/or PRs and issues is fine (and encouraged actually) but adding TODOs in the code should be avoided if we can help it.

Reply:
ack